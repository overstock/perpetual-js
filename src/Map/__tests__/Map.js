import * as Map from '../index';

describe('Map', () => {
  test('emptyMap to return empty Map', () => {
    const map = {
      size: 0,
      root: undefined,
      altered: false,
    };

    expect(Map.emptyMap()).toEqual(map);
  });

  test('makeEmptyMap creates Map with correct size', () => {
    const map = {
      size: 0,
      root: undefined,
      altered: false,
    };

    expect(Map.makeEmptyMap(0)).toEqual(map);
  });

  describe('createNewMap', () => {
    test('if typeof value is not object, throws error', () => {
      expect(Map.createNewMap).toThrow(Error);
    });

    test('return Map if isArray', () => {
      const value = [['test', 'test']];
      expect(Map.createNewMap(value)).toMatchSnapshot();
    });

    test('returns Map if is Object', () => {
      const value = { test: 'test' };
      expect(Map.createNewMap(value)).toMatchSnapshot();
    });
  });

  describe('Mep', () => {
    const value = { test: 'test' };

    test('returns empty map with empty value', () => {
      expect(Map.default()).toMatchSnapshot();
    });

    test('returns new Map with value', () => {
      expect(Map.default(value)).toMatchSnapshot();
    });
  });
});
