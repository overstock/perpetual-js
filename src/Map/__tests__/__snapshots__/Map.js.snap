// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Mep returns empty map with empty value 1`] = `
Map {
  "altered": false,
  "root": undefined,
  "size": 0,
}
`;

exports[`Map Mep returns new Map with value 1`] = `
Map {
  "altered": false,
  "root": ArrayMapNode {
    "entries": Array [
      Array [
        "test",
        "test",
      ],
    ],
  },
  "size": 1,
}
`;

exports[`Map createNewMap return Map if isArray 1`] = `
Map {
  "altered": false,
  "root": ArrayMapNode {
    "entries": Array [
      Array [
        "test",
        "test",
      ],
    ],
  },
  "size": 1,
}
`;

exports[`Map createNewMap returns Map if is Object 1`] = `
Map {
  "altered": false,
  "root": ArrayMapNode {
    "entries": Array [
      Array [
        "test",
        "test",
      ],
    ],
  },
  "size": 1,
}
`;
